### PR TITLE
Add speed-up values to 5.2 whats new

### PR DESCRIPTION
**Pull request by maxnoe**
### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/whatsnew/5.2.rst [hunk 1]
```diff
@@ -116,6 +116,9 @@ Performance Improvements
 To help speed up coordinate transformations, several performance improvements
 were implemented, mainly concerning validity checks in ``Angle`` and
 how ``FrameAttributes``  are accessed.
+Performance improvements will vary between different tasks, e.g. the speedup of
+transforming ``AltAz`` coordinates to ``ICRS`` is 5-10 % depending on the number
```
Comment (eerovaher) [inline docs/whatsnew/5.2.rst:120]: Differently from the change log the What's New page can contain hyperlinks, so there is no need to hold back:
```suggestion
transforming :class:`~astropy.coordinates.AltAz` coordinates to :class:`~astropy.coordinates.ICRS` is 5-10 % depending on the number
```

**Comment by eerovaher** [comment]
There's a line that is a bit too long and the commits should be squashed, but the phrasing looks good enough to merge.

File: docs/whatsnew/5.2.rst [hunk 2]
```diff
@@ -116,6 +116,9 @@ Performance Improvements
 To help speed up coordinate transformations, several performance improvements
 were implemented, mainly concerning validity checks in ``Angle`` and
 how ``FrameAttributes``  are accessed.
+Performance improvements will vary between different tasks, e.g. the speedup of
```
Comment (pllim) [inline docs/whatsnew/5.2.rst:119]: Before you squash... 😸 
```suggestion
Performance improvements will vary between different tasks, e.g., the speedup of
```

**Comment by pllim** [comment]
@astrofrog , is it too late for v5.2 ?

File: docs/whatsnew/5.2.rst (hunk 2 above)
Comment (dhomeier) [reply docs/whatsnew/5.2.rst:119]: 🤔

**Comment by astrofrog** [comment]
No, changes/improvements to docs are fine during the RC stage.

**Comment by pllim** [comment]
@maxnoe , are you interested to wrap this up? If not I can take over, pls lemme know. Thanks!

**Comment by maxnoe** [comment]
Sorry, yes, will squash

**Review by pllim — APPROVED** [review]
Thanks, all!

**Comment by pllim** [comment]
@eerovaher and @dhomeier , do you want one last look or can I merge as-is? Thanks!

**Comment by dhomeier** [comment]
I did not understand the grammar of a trailing comma after `for example`, but maybe that's AE vs. BE idiom.

**Comment by pllim** [comment]
ApJ used to enforce trailing comma for `i.e.` and `e.g.`. Lemme double check.

**Comment by pllim** [comment]
Well, I cannot search effectively with their new UI but they do use trailing comma after `e.g.` in their own style guide narrative, so I guess it still stands: https://journals.aas.org/aas-style-guide/

File: docs/whatsnew/5.2.rst [hunk 3]
```diff
@@ -116,6 +116,9 @@ Performance Improvements
 To help speed up coordinate transformations, several performance improvements
 were implemented, mainly concerning validity checks in ``Angle`` and
 how ``FrameAttributes``  are accessed.
+Performance improvements will vary between different tasks, e.g., the speedup of
+transforming :class:`~astropy.coordinates.AltAz` coordinates to :class:`~astropy.coordinates.ICRS` is 5-10 % depending on the number
```
[the remark this answers is not in the document]
Comment (pllim) [reply docs/whatsnew/5.2.rst:120]: @maxnoe , do you want to address the line length?

**Comment by dhomeier** [comment]
Yea, it's an Americanism https://en.wiktionary.org/wiki/e.g.#Usage_notes

File: docs/whatsnew/5.2.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxnoe) [reply docs/whatsnew/5.2.rst]: done

**Comment by dhomeier** [comment]
Humm,
> /Users/runner/work/astropy/astropy/.tox/.tox/bin/python: No module named tox

both on Actions and CircleCI??

**Comment by maxnoe** [comment]
And both here and on the other PR... seems to be some general issue.

**Review by eerovaher — APPROVED** [review]

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**